pkg/cmd/build/util.go

```diff
@@ -117,6 +125,24 @@ func buildAppConfigs(o *builders.Options, stack *stack.Stack) (map[string]inputs
 	return appConfigs, nil
 }
 
+func getWorkspace(stackName string) (*apisworkspace.Workspace, error) {
+	wsOperator, err := workspace.NewDefaultOperator()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace operator failed, %w", err)
+	}
+	// stack name should be same as the workspace name
+	ws, err := wsOperator.GetWorkspace(stackName)
+	// allow empty workspace
+	if errors.Is(err, workspace.ErrWorkspaceNotExist) {
```

Comment:
what will happen if workspace is nil? Seems `AppConfigurationGenerator` returns err if workspace is nil

Reply:
There is no limit of the empty workspace is because if there are other app configuration schemas which are developed by kusion's users, the workspace may not be necessary. (Just consider an independent developer, maybe he/she will declare all his/her configurations in a new app configuration schema.) Hence, in pkg cmd, if the workspace is empty, there used to be no error returned.
But after revisiting the code, the operation happens under the condition using `appConfigurationBuilder`. It means using kusion‘s own `AppConfiguration` schema, which asks for non-nil workspace. So I should delete the special error handling. You give me a great vetting, thanks!